focus-android/app/src/main/java/org/mozilla/focus/activity/MainActivity.kt

```diff
@@ -272,6 +272,26 @@ open class MainActivity : LocaleAwareAppCompatActivity() {
         super.onNewIntent(unsafeIntent)
     }
 
+    private fun handleAppRestoreFromBackground(intent: SafeIntent) {
+        when (components.appStore.state.screen) {
+            is Screen.Settings -> components.appStore.dispatch(
+                AppAction.OpenSettings(
+                    page =
+                    (components.appStore.state.screen as Screen.Settings).page,
+                ),
+            )
+            is Screen.SitePermissionOptionsScreen -> components.appStore.dispatch(
+                AppAction.OpenSitePermissionOptionsScreen(
+                    sitePermission =
+                    (components.appStore.state.screen as Screen.SitePermissionOptionsScreen).sitePermission,
+                ),
+            )
```

Comment:
Because doesn't make sense for me to do this check when the app is not in the background (savedInstanceState == null) the app doesn't have a screen state and when the lockscreen is shown (state.screen is Lock).

Reply:
Thanks for the clarification!